packages/core/test/transient.test.ts

```diff
@@ -629,4 +629,45 @@ describe('transient states (eventless transitions)', () => {
 
     service.send('ADD');
   });
+
+  it("shouldn't crash when invoking a machine with initial transient transition depending on custom data", () => {
```

Comment:
This is not the best test as @davidkpiano's proposed workaround [here](https://github.com/davidkpiano/xstate/issues/1277#issuecomment-653673951) is valid (to some extent). One should not rely on undefined `context` - if `TContext` parameter is configured then `.context` should satisfy that constraint at all times and thus expressions (like guards) should not fail due to some properties being missed.

However, I don't know what other observable behavior to test at XState layer. The other added test (`@xstate/react`) is much better: https://github.com/davidkpiano/xstate/pull/1285/files#diff-326fee0dc0480200270263670a5d51bfR608

Reply:
@davidkpiano do you think this test here is worth keeping?

Reply:
I think so, yes.